src/material/expansion/testing/expansion-harness.ts

```diff
@@ -0,0 +1,140 @@
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {ComponentHarness, HarnessLoader, HarnessPredicate} from '@angular/cdk/testing';
+import {ExpansionPanelHarnessFilters} from './expansion-harness-filters';
+
+const EXPANSION_PANEL_CONTENT_SELECTOR = '.mat-expansion-panel-content';
+
+/** Harness for interacting with a standard mat-expansion-panel in tests. */
+export class MatExpansionPanelHarness extends ComponentHarness {
+  static hostSelector = '.mat-expansion-panel';
+
+  private _header = this.locatorFor('.mat-expansion-panel-header');
+  private _title = this.locatorForOptional('.mat-expansion-panel-header-title');
+  private _description = this.locatorForOptional('.mat-expansion-panel-header-description');
+  private _expansionIndicator = this.locatorForOptional('.mat-expansion-indicator');
+  private _content = this.locatorFor(EXPANSION_PANEL_CONTENT_SELECTOR);
+
+  /**
+   * Gets a `HarnessPredicate` that can be used to search for an expansion-panel
+   * with specific attributes.
+   * @param options Options for narrowing the search:
+   *   - `title` finds an expansion-panel with a specific title text.
+   *   - `description` finds an expansion-panel with a specific description text.
+   *   - `expanded` finds an expansion-panel that is currently expanded.
+   *   - `disabled` finds an expansion-panel that is disabled.
+   * @return a `HarnessPredicate` configured with the given options.
+   */
+  static with(options: ExpansionPanelHarnessFilters = {}):
+      HarnessPredicate<MatExpansionPanelHarness> {
+    return new HarnessPredicate(MatExpansionPanelHarness, options)
+        .addOption(
+            'title', options.title,
+            async (harness, title) =>
+                this._matchesStringNullableOption(await harness.getTitle(), title))
+        .addOption(
+            'description', options.description,
+            async (harness, description) =>
+                this._matchesStringNullableOption(await harness.getDescription(), description))
+        .addOption(
+            'content', options.content,
+            (harness, content) => HarnessPredicate.stringMatches(harness.getTextContent(), content))
+        .addOption(
+            'expanded', options.expanded,
+            async (harness, expanded) => (await harness.isExpanded()) === expanded)
+        .addOption(
+            'disabled', options.disabled,
+            async (harness, disabled) => (await harness.isDisabled()) === disabled);
+  }
+
+  /** Whether the panel is expanded. */
+  async isExpanded(): Promise<boolean> {
+    return (await this.host()).hasClass('mat-expanded');
+  }
+
+  /**
+   * Gets the title text of the panel.
+   * @returns Title text or `null` if no title is set up.
+   */
+  async getTitle(): Promise<string|null> {
+    const titleEl = await this._title();
+    return titleEl ? titleEl.text() : null;
```

Comment:
Is it important to differentiate between `null` and `''`? If not we could just say `''` all the time so the user doesn't have to think about it

Reply:
Yes, it's important to differentiate. Null means that there is no title, while otherwise the title is just empty. e.g. `<mat-panel-title><mat-icon>...</mat-icon></mat-panel-title>` (as an example).

This should be clear from the method description.

Reply:
Ok, I was wondering why an empty title would look different than no title, but an icon is a good example. It's not empty, it just has no _text_ content

Reply:
:+1: let me know if you think that the JSdoc can be somehow improved.